### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by frankban**
This is an attempt of fixing https://github.com/frankban/quicktest/issues/51

I have stolen some sentences from the excellent blog post at https://rogpeppe.wordpress.com/2019/05/15/quicktest-wrap-testing-t-for-fun-and-profit/

I am not entirely satisfied by the doc organization and repetitions, but some of that is due to godoc lack of many options for structuring text in chapters/sections.

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -22,8 +22,183 @@ instance:
...
+    c.Check(someValue, qt.Equals, wantValue)
+
 The library provides some base checkers like Equals, DeepEquals, Matches,
 ErrorMatches, IsNil and others. More can be added by implementing the Checker
-interface.
+interface. Here is a list of checkers already included in the package.
+
+Equals
+
+Equals is a Checker checking equality of two comparable values.
```
Comment (rogpeppe) [inline doc.go]: Equals checks that two values are equal, as compared with Go's == operator.

?

File: doc.go [hunk 2]
```diff
@@ -22,8 +22,183 @@ instance:
...
+
+    c.Assert(answer, qt.Equals, 42)
+
+Note that the following will fail:
+
+    c.Assert((*sometype)(nil), qt.Equals, nil)
+
+Use the IsNil checker below for this kind of nil check.
+
+CmpEquals
```
Comment (rogpeppe) [inline doc.go]: Maybe put this after DeepEquals, as that's much more common to use?
(alternatively you could sort them all alphabetically)

File: doc.go [hunk 3]
```diff
@@ -22,8 +22,183 @@ instance:
...
+according to the provided compare options. See DeepEquals as an example of
+such a checker, commonly used when no compare options are required.
+For instance:
+
+    c.Assert(list, qt.CmpEquals(cmpopts.SortSlices), []int{42, 47})
+    c.Assert(got, qt.CmpEquals(), []int{42, 47}) // Same as qt.DeepEquals.
+
+DeepEquals
+
+DeepEquals is a Checker deeply checking equality of two arbitrary values.
```
Comment (rogpeppe) [inline doc.go]: Maybe mention the cmp package here, and note that it doesn't allow unexported fields by default, as that's a common gotcha when moving from other testing frameworks?

File: doc.go [hunk 4]
```diff
@@ -22,8 +22,183 @@ instance:
...
+For instance:
+
+    c.Assert(got, qt.DeepEquals, []int{42, 47})
+
+Matches
+
+Matches is a Checker checking that the provided string or fmt.Stringer
```
Comment (rogpeppe) [inline doc.go]:
```
Matches checks that a string or result of calling the String method (if the value implements
fmt.Stringer) matches the provided regular expression.
```
?

File: doc.go [hunk 5]
```diff
@@ -22,8 +22,183 @@ instance:
...
+matches the provided regular expression pattern.
+For instance:
+
+    c.Assert("these are the voyages", qt.Matches, "these are .*")
```
Comment (rogpeppe) [inline doc.go]: maybe use backquotes for the regexp, as that's often good practice?

File: doc.go [hunk 6]
```diff
@@ -22,8 +22,183 @@ instance:
...
+    c.Assert(net.ParseIP("1.2.3.4"), qt.Matches, "1.*")
+
+ErrorMatches
+
+ErrorMatches is a Checker checking that the provided value is an error whose
```
Comment (rogpeppe) [inline doc.go]:
```
ErrorMatches checks that the provided value is an error... etc
```
?

I think that phrasing reads a little less awkwardly. Also suggest similar changes below.

File: doc.go [hunk 7]
```diff
@@ -22,8 +22,183 @@ instance:
...
+On failure it prints the error from the first index that failed.
+
+For example:
+
+	c.Assert([]int{3, 5, 8}, qt.All(qt.Not(qt.Equals)), 0)
+	c.Assert([][]string{{"a", "b"}, {"a", "b"}}, qt.All(qt.DeepEquals), []string{"c", "d"})
+
+See also Any and Contains.
+
+Deferred execution
```
Comment (rogpeppe) [inline doc.go]: Maybe put this before all the checkers?

File: doc.go [hunk 8]
```diff
@@ -22,8 +22,183 @@ instance:
...
+
+Quicktest provides the ability to defer the execution of functions that will be
+run in last-in, first-out order when the test completes. This is often useful
```
Comment (rogpeppe) [inline doc.go]: Maybe keep the "last-in, first-out" thing for a later sentence?

```
Quicktest provides the ability to defer the execution of functions that will be
run when the test completes. This is often useful.... etc.
The functions will be run in last-in, first-out order.
```

?

File: doc.go (hunk 1 above)
Comment (frankban) [reply doc.go]: Done.

File: doc.go (hunk 2 above)
Comment (frankban) [reply doc.go]: Done.

File: doc.go (hunk 3 above)
Comment (frankban) [reply doc.go]: Done.

File: doc.go (hunk 4 above)
Comment (frankban) [reply doc.go]: Done.

File: doc.go (hunk 5 above)
Comment (frankban) [reply doc.go]: Done.

File: doc.go (hunk 6 above)
Comment (frankban) [reply doc.go]: Done.

File: doc.go (hunk 7 above)
Comment (frankban) [reply doc.go]: Done.

File: doc.go (hunk 8 above)
Comment (frankban) [reply doc.go]: Done.

**Pull request merged**